File: src/MaksIT.CertsUI/Models/APIKeys/Search/SearchApiKeyEntityScopeRequest.cs
using System.Linq.Expressions;
using MaksIT.Core.Webapi.Models;
using MaksIT.CertsUI.Engine.Dto.Identity;

namespace MaksIT.CertsUI.Models.APIKeys.Search;

public class SearchApiKeyEntityScopeRequest : PagedRequest {
  /// <summary>
  /// Optional filter by parent API key ID.
  /// </summary>
  public Guid? ApiKeyId { get; set; }

  public Expression<Func<ApiKeyEntityScopeDto, bool>>? BuildFilterExpression() =>
    BuildFilterExpression<ApiKeyEntityScopeDto>(Filters);
}

File: src/MaksIT.WebUI/src/models/apiKey/search/SearchApiKeyEntityScopeRequest.ts
import { object, string, type ZodType } from 'zod'
import { PagedRequest, PagedRequestSchema } from '@maks-it.com/webui-contracts'

/** Mirrors <c>MaksIT.CertsUI.Models.APIKeys.Search.SearchApiKeyEntityScopeRequest</c>. */
export interface SearchApiKeyEntityScopeRequest extends PagedRequest {
  apiKeyId?: string
}

export const SearchApiKeyEntityScopeRequestSchema: ZodType<SearchApiKeyEntityScopeRequest> = PagedRequestSchema.and(
  object({
    apiKeyId: string().optional(),
  })
)
